File: cloud-starter-kit-admin/kit-utils/cdk/python-example/stacks/example_stack.py
from aws_cdk import (
    Stack,
    aws_ec2 as ec2,
)
from constructs import Construct


class ExampleStack(Stack):

    def __init__(
        self, scope: Construct, construct_id: str, params: map, **kwargs
    ) -> None:
        super().__init__(scope, construct_id, **kwargs)

        vpc = ec2.Vpc(
            self,
            "Vpc",
            vpc_name="ask-vpc-with-nat",
            nat_gateways=int(params["natCount"]),
            max_azs=3,
        )
        self.export_value(exported_value=vpc.vpc_id, name="CskVpcIdWithNat")

File: cloud-starter-kit-admin/kit-utils/cdk/js-example/lib/example-stack.js
const { Stack } = require('aws-cdk-lib');
const ec2 = require('aws-cdk-lib/aws-ec2');

class ExampleStack extends Stack {
  /**
   *
   * @param {Construct} scope
   * @param {string} id
   * @param {StackProps=} props
   */
  constructor(scope, id, props) {
    super(scope, id, props);

    const vpc = ec2.Vpc.fromLookup(this, 'Vpc', {
      vpcId: props.params["vpcId"],
    });

    //do your thing

  }
}


module.exports = { ExampleStack }
